### ISSUE TITLE
ResourceTable attributes

### ISSUE THREAD

**Issue by Eastrall** [Eastrall/Hellion#54]
We have loaded a lot of data to the server already. Something that can be intersting to do is to reduce the code size.

For example this file: https://github.com/Eastrall/Hellion/blob/develop/src/Hellion.Core/Structures/ItemData.cs
and this one: https://github.com/Eastrall/Hellion/blob/develop/src/Hellion.Core/Structures/SkillData.cs

are almost the same.

We could think to do something like the `Newtonsoft.Json` library, which is to have flag our structure fields with attributes. Just like this: 

```c#
public struct ItemData
{
    [MyAttribute(Name = "dwId")]
    public int Id;

    // etc ...
}
```

If we want to load data from a file, we will just call a method like:
```c#
IEnumerable<ItemData> itemsData = ResourceTable.Parse<ItemData>("propItem.txt");
```

The `ResourceTable.Parse` will process some reflection the `ItemData` to find out all the flaged fields with the attribute `MyAttribute`.

This will result on a more elevated cost than the solution we have now. But notice that this type of code will be executed when the emulator is loading data. No live loading during the server is up and welcomes players.

What do you think?

**Comment by raventh1984** [comment]
You can remove an lot of skilldata.cs since its not beeing used for skills.
Num,
PackMax,
Cost,
Usable, ItemSex etc.

**Comment by Eastrall** [comment]
Yes, that's why I want to use an attribute system so define only what we want:

ItemData:
```c#
public struct ItemData
{
     [MyAttribute(Name = "dwId")]
     public int Id;

     [MyAttribute(Name = "dwCost")]
     public int cost;
}
```

SkillData
```c#
public struct SkillData
{
     [MyAttribute(Name = "dwId")]
     public int Id;

     // Other properties related with SkillData
}
```
